ddtrace/internal/sampling.py

```diff
@@ -1,18 +1,29 @@
 import re
+from typing import Optional
 from typing import TYPE_CHECKING
 
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_MAX_PER_SEC
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_MECHANISM
+from ddtrace.constants import _SINGLE_SPAN_SAMPLING_RATE
+from ddtrace.internal.glob_matching import GlobMatcher
 from ddtrace.internal.logger import get_logger
 
+from .rate_limiter import RateLimiter
+
 
 log = get_logger(__name__)
 
 if TYPE_CHECKING:
     from typing import Dict
-    from typing import Optional
     from typing import Text
 
     from ddtrace.context import Context
 
+    from ..span import Span
+
+KNUTH_FACTOR = 1111111111111111111
```

Comment:
Can we add a comment here. It's not immediately clear what the `KNUTH_FACTOR` is or why it has that specific value

Reply:
Good call out, I actually don't know why it has that specific value either. I thought that it should be a prime value, but it doesn't appear to be. https://lowrey.me/exploring-knuths-multiplicative-hash-2/ @brettlangdon do you know why we use this knuth factor in our normal sampling? I don't understand the comment above: https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/sampler.py#L50

Reply:
> Because it is prime and "big" (high highest order bit), which makes the hashing better distributed.